File: ocaml/005/main.ml
(*
    ProjectEuler 005
    OCaml / main.ml
    copyright (c) 2015 Susisu

    OCaml 4.02.1
    `ocamlc main.ml -o bin/main`
*)

let rec gcd a b =
    if a < b then
        gcd b a
    else if b = 0 then
        a
    else
        gcd b (a mod b);;

let rec calc n p =
    if n <= 20 then
        calc (n + 1) (p * n / gcd p n)
    else
        p;;

let main () = begin
    print_int (calc 1 1);
    print_newline ()
end;;

main ();;

File: rust/005/main.rs
/*
    ProjectEuler 005
    Rust / main.rs
    copyright (c) 2015 Susisu

    Rust 1.3.0
    `rustc main.rs -o bin/main`
*/

fn main() {
    let mut n = 1;
    for i in 1..20 {
        n /= gcd(n, i);
        n *= i;
    }
    println!("{}", n);
}

fn gcd(m: i32, n: i32) -> i32 {
    if m < n {
        gcd(n, m)
    }
    else if n > 0 {
        gcd(n, m % n)
    }
    else {
        m
    }
}
